dotCMS/src/integration-test/java/com/dotcms/publishing/job/SiteSearchJobImplTest.java

```diff
@@ -448,74 +454,441 @@ public void Test_Incremental_Job_Test_Pages_Are_Found_Create_And_Publish_New_Pag
+            Mockito.when(context.getJobDetail()).thenReturn(jobDetail);
+            Mockito.when(context.getFireTime()).thenReturn(new Date());
+            final SiteSearchJobImpl impl = new SiteSearchJobImpl();
+            impl.run(context);
 
-            final String uuid = UUIDGenerator.generateUuid();
+            final List<String> indicesAfterTest = siteSearchAPI.listIndices();
+            Assert.assertFalse(indicesAfterTest.isEmpty());
+            final String newIndexName = indicesAfterTest.get(0);
 
-            final Template template = new TemplateDataGen()
-                    .withContainer(container.getIdentifier(), uuid)
-                    .nextPersisted();
+            SiteSearchResults searchResults = siteSearchAPI.search(newIndexName, "catalina", 0, 10);
+            Assert.assertTrue("Content in default Language gives results", searchResults.getTotalResults() >= 1);
+            Assert.assertEquals(page.getTitle(), searchResults.getResults().get(0).getTitle());
+            Assert.assertEquals(page.getLanguageId(), searchResults.getResults().get(0).getLanguage());
 
-            TemplateDataGen.publish(template);
 
-            HTMLPageAsset page = new HTMLPageDataGen(folder, template).languageId(lang1.getId())
-                    .nextPersisted();
+            searchResults = siteSearchAPI.search(newIndexName, "catherine", 0, 10);
+            Assert.assertTrue("Content in second Language gives results", searchResults.getTotalResults() >= 1);
+            Assert.assertEquals(page.getTitle(), searchResults.getResults().get(0).getTitle());
+            Assert.assertEquals(page.getLanguageId(), searchResults.getResults().get(0).getLanguage());
+
+
+        } finally {
+            Config.setProperty("DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE",
+                    defaultContentToDefaultLangOriginalValue);
+        }
+    }
 
-            HTMLPageDataGen.publish(page);
+    private void createNewVersionAndPublishExistingEmployeeContent(Language secondLang,
+            Contentlet contentletDefaultLang, String firstName) throws DotDataException, DotSecurityException {
+        Contentlet contentletSecondLang = contentletAPI
+                .find(contentletDefaultLang.getInode(), systemUser, false);
+        contentletSecondLang.setInode("");
+        contentletSecondLang.setLanguageId(secondLang.getId());
+        contentletSecondLang.setStringProperty("firstName", firstName);
+        contentletSecondLang = contentletAPI.checkin(contentletSecondLang, systemUser, false);
+        ContentletDataGen.publish(contentletSecondLang);
+    }
 
-            final MultiTree multiTree = new MultiTree(page.getIdentifier(),
-                    container.getIdentifier(),
-                    contentletLang1.getIdentifier(), getDotParserContainerUUID(uuid), 0);
+    private Contentlet createAndPublishEmployeeContent(Host site, Language language,
+            String firstName) throws DotDataException, DotSecurityException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'firstName' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5303548)